modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dataSetsPermissions.spec.ts

```diff
@@ -0,0 +1,560 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page, expect, mergeTests} from '@playwright/test';
+
+import {dataApiHelpersTest} from '../../../../fixtures/dataApiHelpersTest';
+import {featureFlagsTest} from '../../../../fixtures/featureFlagsTest';
+import {loginTest} from '../../../../fixtures/loginTest';
+import {liferayConfig} from '../../../../liferay.config';
+import getRandomString from '../../../../utils/getRandomString';
+import performLogin, {
+	performLogout,
+	performUserSwitch,
+} from '../../../../utils/performLogin';
+import {waitForAlert} from '../../../../utils/waitForAlert';
+import {dataSetManagerApiHelpersTest} from '../../fixtures/dataSetManagerApiHelpersTest';
+import {setupUserRoleAndLoginAsUser} from '../../utils/setupUserRoleAndLoginAsUser';
+import {dataSetsPageTest} from './fixtures/dataSetsPageTest';
+
+export const test = mergeTests(
+	dataApiHelpersTest,
+	dataSetManagerApiHelpersTest,
+	dataSetsPageTest,
+	featureFlagsTest({
+		'LPS-178052': true,
+	}),
+	loginTest()
+);
+
+const createdDataSetERCs = [];
+const createdRoleIds = [];
+const createdUserIds = [];
+const dataSetUserRoleName = `ds_user_${getRandomString()}`;
+
+const blogPostsDataSetConfig = {
+	name: 'BlogPost',
+	restApplication: '/headless-delivery/v1.0',
+	restEndpoint: '/v1.0/sites/{siteId}/blog-postings',
+	restSchema: 'BlogPosting',
+};
+
+async function openActionsDropdown({page, text}: {page: Page; text: string}) {
+	const table: Locator = page.locator('.data-set-content-wrapper');
+
+	const row = table.locator('.dnd-tbody .dnd-tr').filter({
+		has: page.getByText(text, {exact: true}).first(),
+	});
+
+	const actionsButton = row.locator('.cell-item-actions button');
+
+	await expect(actionsButton).toBeInViewport();
+
+	await actionsButton.click();
+}
+
+test.beforeEach(async ({page}) => {
+	if (
```

Comment:
Created task to investigate login/logut issues
https://liferay.atlassian.net/browse/LPD-43289